src/c_kzg_4844.c

```diff
@@ -719,13 +716,14 @@ static void compute_challenge(
     uint8_t *offset = bytes;
 
     /* Copy domain separator */
-    memcpy(offset, FIAT_SHAMIR_PROTOCOL_DOMAIN, 16);
-    offset += 16;
+    memcpy(offset, FIAT_SHAMIR_PROTOCOL_DOMAIN, BYTES_PER_DOMAIN);
```

Comment:
Perhaps instead of BYTES_PER_DOMAIN we can use strlen() here?



Also, usually "domain separator" is the name of the entire header before the actual data, whereas here `BYTES_PER_DOMAIN` only refers to that string.

Reply:
So I renamed this to `DOMAIN_STR_LENGTH`. I tried out using `strlen` but wasn't really a fan. One concrete reason, I would like to be able to use this value (the str length) in the new `DOMAIN_SEPARATOR_LENGTH` constant. If we use `strlen` at runtime this isn't available.

Reply:
Another alternative just for completeness is to turn the string into a constant array, and then use `sizeof(str) - 1` at compile time. IIRC that works as well.

Reply:
Hmm maybe. Not really a fan of the `-1` part though.